core/src/main/java/kafka/log/remote/RemoteLogOffsetReader.java

```diff
@@ -61,17 +60,18 @@ public RemoteLogOffsetReader(RemoteLogManager rlm,
 
     @Override
     public Void call() throws Exception {
-        Either<Exception, Option<FileRecords.TimestampAndOffset>> result;
+        OffsetResultHolder.FileRecordsOrError result;
         try {
             // If it is not found in remote storage, then search in the local storage starting with local log start offset.
             Option<FileRecords.TimestampAndOffset> timestampAndOffsetOpt =
                     OptionConverters.toScala(rlm.findOffsetByTimestamp(tp, timestamp, startingOffset, leaderEpochCache))
                     .orElse(searchInLocalLog::get);
-            result = Right.apply(timestampAndOffsetOpt);
+            result = new OffsetResultHolder.FileRecordsOrError(Optional.empty(),
+                    timestampAndOffsetOpt.isDefined() ? Optional.of(timestampAndOffsetOpt.get()) : Optional.empty());
```

Comment:
Can't `timestampAndOffsetOpt.isDefined() ? Optional.of(timestampAndOffsetOpt.get()) : Optional.empty()` be simplified to just `timestampAndOffsetOpt`?

Reply:
Still, maybe something like https://www.scala-lang.org/api/2.13.6/scala/jdk/OptionConverters$.html will simplify this?